src/core_plugins/vega/public/vega_view/vega_view.js

```diff
@@ -0,0 +1,283 @@
+  }
+
+  updateVegaSize(view) {
+    // FIXME: for some reason the object is slightly scrollable without this
+    const heightExtraPadding = 6;
+    const width = Math.max(0, this._$container.width() - this._parser.paddingWidth);
+    const height = Math.max(0, this._$container.height() - this._parser.paddingHeight) - heightExtraPadding;
+    if (view.width() !== width || view.height() !== height) {
+
+      // fixme: remove
+      if (!view.____id) view.____id = Math.ceil(Math.random() * 10000);
+      console.log(`new Vega size #${view.____id} = ${width}x${height}`);
+
+      view.width(width).height(height);
+      return true;
+    }
+    return false;
+  }
+
+  async _initRawVega() {
+    // In some cases, Vega may be initialized twice... TBD
+    if (!this._$container) return;
+
+    const view = new vega.View(vega.parse(this._parser.spec), this._vegaViewConfig);
+    VegaView.setDebugValues(view, this._parser.spec, this._parser.vlspec);
+
+    view.warn = this.onWarn.bind(this);
+    view.error = this.onError.bind(this);
+    if (this._parser.useResize) this.updateVegaSize(view);
+    view.initialize(this._$container.get(0), this._$controls.get(0));
+
+    if (this._parser.useHover) view.hover();
+
+    this._addDestroyHandler(() => {
+      this._view = null;
+      view.finalize();
+    });
+
+    await view.runAsync();
+    this._view = view;
+  }
+
+  async _initLeafletVega() {
+    const mapConfig = this._parser.mapConfig;
+    const useDefaultMap = mapConfig.mapStyle !== false;
+
+    let limits;
+    let url;
+    let baseLayer;
+
+    if (useDefaultMap) {
+      const tmsService = await this._serviceSettings.getTMSService();
+      // FIXME: In some cases, Vega may be initialized twice, e.g. after awaiting... TBD
+      if (!this._$container) return;
+
+      url = tmsService.getUrl();
+      limits = tmsService.getTMSOptions();
+    } else {
+      limits = { minZoom: 0, maxZoom: 25 };
+    }
+
+    const validate = (name, value, dflt, min, max) => {
+      if (value === undefined) {
+        value = dflt;
+      } else if (value < min) {
+        this.onWarn(`Resetting ${name} to ${min}`);
+        value = min;
+      } else if (value > max) {
+        this.onWarn(`Resetting ${name} to ${max}`);
+        value = max;
+      }
+      return value;
+    };
+
+    let minZoom = validate('minZoom', mapConfig.minZoom, limits.minZoom, limits.minZoom, limits.maxZoom);
+    let maxZoom = validate('maxZoom', mapConfig.maxZoom, limits.maxZoom, limits.minZoom, limits.maxZoom);
+    if (minZoom > maxZoom) {
+      this.onWarn('minZoom and maxZoom have been swapped');
+      [minZoom, maxZoom] = [maxZoom, minZoom];
+    }
+    const zoom = validate('zoom', mapConfig.zoom, 2, minZoom, maxZoom);
+
+    // let maxBounds = null;
+    // if (mapConfig.maxBounds) {
+    //   const b = mapConfig.maxBounds;
+    //   maxBounds = L.latLngBounds(L.latLng(b[1], b[0]), L.latLng(b[3], b[2]));
+    // }
+
+    const map = L.map(this._$container.get(0), {
```

Comment:
Map support needs to be removed until there is a solution for VegaPlugin using KibanaMap. Leaflet dependency should not bleed in here. We can extend KibanaMap to include the required events to monitor zoom/changes. Already it publishes some changes in map-extent, which may even be sufficient: https://github.com/thomasneirynck/kibana/blob/66a6fbfdab9a8134ed7d8a72996abb4cd68d9272/src/core_plugins/tile_map/public/kibana_map.js#L113-L120